### ISSUE TITLE
Different numbers of common.genes using the same reference and de.n

### ISSUE THREAD

**Issue by r-panero** [LTLA/SingleR#142]
Hello,
If I understood it correctly SingleR uses the union of the de.n genes among the pairwise comparisons of the reference labels.
I tried to run SingleR using different test samples with the same reference and the same de.n number with the classic method but the number of the common.genes is not the same. 
Since the reference and the de.n number are the same I should get the same number of common.gene, is it correct?
Thank you.

**Comment by LTLA** [comment]
Depends on the gene annotation in your test samples.

**SingleR** starts by taking the intersection of genes in the test and training datasets. After all, there's little point drawing inferences about a test dataset for genes that don't even exist in the training dataset. The markers are then identified within this intersection. If your test samples have the same gene annotations, then yes, the markers should also be the same. We do it this way because otherwise we could otherwise end up in a situation where we identify markers from the training set... and none of those markers exist in the test.

Now, in practice, different test samples have different gene annotations - just because, you know, biologists and toothbrushes and gene names and all that. This means that the intersection may be different and the markers may be different. The `common.genes` is simply a union of all the pairwise markers so that may differ as well.

For historical reasons, the reference annotation is reported in gene symbols. This has caused all sorts of amusing problems where people accidentally use mouse test datasets with human annotations (or vice versa); the subsequent **SingleR** results are weird because they are based on the intersection of genes that happen to have all-caps symbols in both species, e.g., lncRNAs and predicted transcripts. IMO it's much safer to work with Ensembl IDs.
